builder/vmware/common/driver_parser_test.go

```diff
@@ -0,0 +1,494 @@
+		t.Errorf("Expected %#v, received %#v", result_0, result)
+	}
+
+	test_1 := "this is a straight-up line with a newline\n"
+	result_1 := test_1
+
+	result = uncommentFromString(test_1)
+	if result != result_1 {
+		t.Errorf("Expected %#v, received %#v", result_1, result)
+	}
+
+	test_2 := "this line has a comment # at its end"
+	result_2 := "this line has a comment "
+
+	result = uncommentFromString(test_2)
+	if result != result_2 {
+		t.Errorf("Expected %#v, received %#v", result_2, result)
+	}
+
+	test_3 := "# this whole line is commented"
+	result_3 := ""
+
+	result = uncommentFromString(test_3)
+	if result != result_3 {
+		t.Errorf("Expected %#v, received %#v", result_3, result)
+	}
+
+	test_4 := "this\nhas\nmultiple\nlines"
+	result_4 := test_4
+
+	result = uncommentFromString(test_4)
+	if result != result_4 {
+		t.Errorf("Expected %#v, received %#v", result_4, result)
+	}
+
+	test_5 := "this only has\n# one line"
+	result_5 := "this only has\n"
+	result = uncommentFromString(test_5)
+	if result != result_5 {
+		t.Errorf("Expected %#v, received %#v", result_5, result)
+	}
+
+	test_6 := "this is\npartially # commented"
+	result_6 := "this is\npartially "
+
+	result = uncommentFromString(test_6)
+	if result != result_6 {
+		t.Errorf("Expected %#v, received %#v", result_6, result)
+	}
+
+	test_7 := "this # has\nmultiple # lines\ncommented # out"
+	result_7 := "this \nmultiple \ncommented "
+
+	result = uncommentFromString(test_7)
+	if result != result_7 {
+		t.Errorf("Expected %#v, received %#v", result_7, result)
+	}
+}
+
+func tokenizeDhcpConfigFromString(s string) []string {
+	inCh, eos := consumeString(s)
+	out, eoc := tokenizeDhcpConfig(eos, inCh)
+
+	result := make([]string, 0)
+	for reading := true; reading; {
+		select {
+		case <-eoc:
+			reading = false
+		case item, ok := <-out:
+			if ok {
+				result = append(result, item)
+			}
+		}
+	}
+	close(out)
+	return result
+}
+
+func TestParserTokenizeDhcp(t *testing.T) {
+
+	test_1 := `
+subnet 127.0.0.0 netmask 255.255.255.252 {
+    item 1234 5678;
+	tabbed item-1 1234;
+    quoted item-2 "hola mundo.";
+}
+`
+	expected := []string{
+		"subnet", "127.0.0.0", "netmask", "255.255.255.252", "{",
+		"item", "1234", "5678", ";",
+		"tabbed", "item-1", "1234", ";",
+		"quoted", "item-2", "\"hola mundo.\"", ";",
+		"}",
+	}
+	result := tokenizeDhcpConfigFromString(test_1)
+
+	t.Logf("testing for: %v", expected)
+	t.Logf("checking out: %v", result)
+	if len(result) != len(expected) {
+		t.Fatalf("length of token lists do not match (%d != %d)", len(result), len(expected))
+	}
+
+	for index := range expected {
+		if expected[index] != result[index] {
+			t.Errorf("unexpected token at index %d: %v != %v", index, expected[index], result[index])
+		}
+	}
+}
+
+func consumeTokens(tokes []string) chan string {
+	out := make(chan string)
+	go func() {
+		for _, item := range tokes {
+			out <- item
+		}
+		out <- ";"
+		close(out)
+	}()
+	return out
+}
+
+func TestParserDhcpParameters(t *testing.T) {
```

Comment:
It's not worth a refactor, but for future reference generally when we do these kinds of tests in series, we'll do a "table test" where we create a struct containing all the test cases instead of having to copy/paste identical test logic for different cases. example: https://github.com/hashicorp/packer/blob/master/packer/ui_test.go#L211

Reply:
gotcha.